### PR TITLE
feat: Added transaction types classes for manage the transaction logic

### PR DESCRIPTION
**Pull request by jjhcuadrosIOV**
### Isolate  business logic to independent classes
Added two new classes to manage all the logic related to every token type (ERC20 and ERC721)

### Description

- Added `Transaction` parent class with `cross`, `approve`, and `claim` methods
- Added `ERC20TokenTransaction` extends from `Transaction` and implement all the methods
- Added `ERC721NFTTransaction` extends from `Transaction` and implement all the methods
- Moved all the logic in `transactions.actions.js` to `ERC20TokenTransaction.js` inside the class
- Modified `CrossForm.vue` to initialize an instance of `ERC20TokenTransaction` and use the methods instead of the current implementation.

### How to Test

- Config your `.env` file for chains `RSK Testnet` and `Kovan`
- Run the server

#### Case 1

1. Run the server
```shell
$~ npm run serve
```

2. Transfer some token from `RSK Testnet` to `Kovan` or vice versa

__Expected Result__
- The token should be transferred correctly

### Checklist
- [x] Lint is clean `npm run lint`
- [x] Prettier is passing `npm run prettier`

Signed-off-by: Jonathan Huamani <jonathan.huamani@iovlabs.org>

### CODE REVIEW

**Review by RafilxTenfen — APPROVED** [review]

**Comment by RafilxTenfen** [comment]
Sounds good to me ^^

File: src/components/nftForm/NFTForm.vue [hunk 1]
```diff
@@ -117,16 +117,28 @@ export default {
...
+            modalProps: { message: approvedError.message, title: 'Error checking approved status' },
+          },
         })
         this.isLoading = false
         return
       }
-      const [uriError, response] = await asyncTryCatch(fetch, uri)
+      const [uriError, response] = await asyncTryCatch(
+        fetch,
+        `https://cors-anywhere.herokuapp.com/${uri}`,
```
Comment (marcciosilva) [inline src/components/nftForm/NFTForm.vue:129]: isn't this only needed for local testing?

File: src/modules/transactions/transaction.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    return `0x${Math.ceil(gasPriceParsed).toString(16)}`
+  }
+  async approve() {
+    throw new Error('You are using a transaction method that require override')
+  }
+  async claim() {
+    throw new Error('You are using a transaction method that require override')
+  }
+  async cross() {
+    throw new Error('You are using a transaction method that require override')
```
Comment (marcciosilva) [inline src/modules/transactions/transaction.js:50]: I'd change require for requires

Comment (jjhcuadrosIOV) [reply src/modules/transactions/transaction.js]: Done

File: src/components/nftForm/NFTForm.vue (hunk 1 above)
Comment (jjhcuadrosIOV) [reply src/components/nftForm/NFTForm.vue]: This is necessary for every request to recover the metadata information; I moved the base URL to an environment variable.

**Review by marcciosilva — APPROVED** [review]

**Pull request merged**